js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx

```diff
@@ -13,220 +11,185 @@ interface SplitPanelContainerProps {
-    const firstSize = second ? _size : 100;
 
     // Note: The divider appears after the first panel, so making the first panel 100% wide
     // hides the divider offscreen. To prevent this, we subtract the divider depth.
     if (axis === 'horizontal') {
       firstPaneStyles.minWidth = firstMinSize;
-      firstPaneStyles.width = `calc(${firstSize}% - ${DIVIDER_THICKNESS}px)`;
+      firstPaneStyles.width = `calc(${firstSize / 100} * (100% - ${
+        DIVIDER_THICKNESS + (second ? secondMinSize : 0)
+      }px))`;
     } else {
       firstPaneStyles.minHeight = firstMinSize;
-      firstPaneStyles.height = `calc(${firstSize}% - ${DIVIDER_THICKNESS}px)`;
+      firstPaneStyles.height = `calc(${firstSize / 100} * (100% - ${
+        DIVIDER_THICKNESS + (second ? secondMinSize : 0)
+      }px))`;
     }
 
     return (
-      <Container axis={axis} id="split-panel-container" resizing={resizing}>
+      <Container axis={axis} className="split-panel-container" resizing={resizing}>
         <div className="split-panel" style={firstPaneStyles}>
           {first}
         </div>
         <PanelDivider
           axis={axis}
           resizing={resizing}
-          onSetResizing={(resizing) => this.setState({resizing})}
-          onMove={this.onChangeSize}
+          secondMinSize={secondMinSize}
+          onSetResizing={setResizing}
+          onMove={onChangeSize}
         />
         <div className="split-panel" style={{flex: 1}}>
           {second}
         </div>
       </Container>
     );
-  }
-}
+  },
+);
 
 interface IDividerProps {
   axis: 'horizontal' | 'vertical';
   resizing: boolean;
+  secondMinSize: number;
   onSetResizing: (resizing: boolean) => void;
   onMove: (vw: number) => void;
 }
 
-class PanelDivider extends React.Component<IDividerProps> {
-  ref = React.createRef<any>();
+const PanelDivider = (props: IDividerProps) => {
+  const {axis, resizing, secondMinSize, onSetResizing, onMove} = props;
+  const ref = useRef<HTMLDivElement>(null);
 
-  onMouseDown = (e: React.MouseEvent) => {
+  const onMouseDown = (e: React.MouseEvent) => {
     e.preventDefault();
 
-    this.props.onSetResizing(true);
+    onSetResizing(true);
 
     const onMouseMove = (event: MouseEvent) => {
-      const parent = this.ref.current?.closest('#split-panel-container');
+      const parent = ref.current?.closest('.split-panel-container');
       if (!parent) {
         return;
       }
       const parentRect = parent.getBoundingClientRect();
 
       const firstPanelPercent =
-        this.props.axis === 'horizontal'
-          ? ((event.clientX - parentRect.left) * 100) / parentRect.width
-          : ((event.clientY - parentRect.top) * 100) / parentRect.height;
+        axis === 'horizontal'
+          ? ((event.clientX - parentRect.left) * 100) /
+            (parentRect.width - secondMinSize - DIVIDER_THICKNESS)
+          : ((event.clientY - parentRect.top) * 100) /
+            (parentRect.height - secondMinSize - DIVIDER_THICKNESS);
```

Comment:
Ensures that the divider remains in the correct position with respect to the minimums applied.